backend/src/routes/file.ts

```diff
@@ -187,6 +197,45 @@ export class FileRoutes {
     }
   }
 
+  deleteFile: RequestHandler = async (req: Request, res: Response, next) => {
+    const uuid = req.params.uuid
+    const query: any = req.query
+    const ignoreHigherProducts = 'ignoreHigherProducts' in query && query.ignoreHigherProducts.toLowerCase() === 'true'
+    try {
+      const existingFile = await this.findAnyFile((repo) => repo.findOne(uuid, {relations: ['product', 'site']}))
+      if (!existingFile) return next({status: 422, errors: ['No file matches the provided uuid']})
+      if (!existingFile.volatile) return next({status: 422, errors: ['Forbidden to delete a stable file']})
+      let fileRepo: Repository<RegularFile|ModelFile> = this.fileRepo
+      let visuRepo: Repository<Visualization|ModelVisualization> = this.visualizationRepo
+      if (existingFile.product.id == 'model') {
+        fileRepo = this.modelFileRepo
+        visuRepo = this.modelVisualizationRepo
+      }
+      const higherLevelProductNames = await this.getHigherLevelProducts(existingFile.product)
+      let products = await this.fileRepo.find({where: {
+        site: existingFile.site,
+        measurementDate: existingFile.measurementDate,
+        product: In(higherLevelProductNames)
+      }})
+      if (ignoreHigherProducts || products.length == 0) {
+        await this.deleteFileAndVisualizations(fileRepo, visuRepo, uuid)
+        return res.sendStatus(200)
+      }
```

Comment:
Kääntäisin ehtolauseen toisinpäin, jotta `deleteFileAndVisualizations` funktiota kutsuttaisiin vain yhdessä kohdassa.

Reply:
Joo hyvä idea.